datalad/support/protocol.py

```diff
@@ -0,0 +1,147 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+""" Protocolling  command calls.
+"""
+
+from os import linesep
+
+
+class ProtocolInterface(object):
+    """Interface class for protocols used by the Runner.
+    """
+
+    def __init__(self):
+        pass
+
+    def write_section(self, cmd, time, exception):
+        """Adds a section to the protocol.
+
+        Parameters:
+        -----------
+        cmd: list
+          The actual command and its options/arguments as a list
+        time:
+          Execution time of the command in seconds.
+        exception: Exception
+          The exception raised by the command if any or None otherwise.
+        """
+        pass
+
+    def write_ext_commands(self):
+        """Indicates whether or not the protocol is supposed to include
```

Comment:
bad name since implies that it writes it... may be "can_write_ext_commands" or "writes_ext_commands"?

and with bother with function calls -- make it a `@property`?